File: AI_AGENTS/AI_TESTPLANNER_AGENT/tools/jira_fetcher.py
import requests
from requests.auth import HTTPBasicAuth

def fetch_jira_ticket(jira_id: str, domain: str, email: str, api_token: str):
    """
    Fetches the ticket summary and description from Jira (using API v2 for string descriptions).
    """
    if not domain or not email or not api_token:
        raise ValueError("Jira credentials are incomplete.")
        
    url = f"{domain.rstrip('/')}/rest/api/2/issue/{jira_id}"
    auth = HTTPBasicAuth(email, api_token)
    headers = {
        "Accept": "application/json"
    }

    response = requests.request("GET", url, headers=headers, auth=auth)
    
    if response.status_code != 200:
        raise Exception(f"Failed to fetch Jira ticket {jira_id}: {response.text}")

    data = response.json()
    fields = data.get("fields", {})
    
    summary = fields.get("summary", "")
    description = fields.get("description", "")
    
    return {
        "summary": summary,
        "description": description if description else ""
    }

def test_jira_connection(domain: str, email: str, api_token: str):
    """
    Test connection by fetching current user info
    """
    if not domain or not email or not api_token:
        return False, "Jira credentials are missing."
        
    url = f"{domain.rstrip('/')}/rest/api/2/myself"
    auth = HTTPBasicAuth(email, api_token)
    headers = {"Accept": "application/json"}
    
    try:
        response = requests.request("GET", url, headers=headers, auth=auth, timeout=10)
        if response.status_code == 200:
            return True, "Jira connected successfully."
        return False, f"Jira error: {response.text}"
    except Exception as e:
        return False, f"Jira error: {str(e)}"

File: AI_AGENTS/AI_TESTPLANNER_AGENT/backend/src/tools/jiraFetcher.ts
import axios from 'axios';

export async function fetchJiraTicket(jiraId: string, url: string, email: string, token: string): Promise<any> {
    if (!url || !email || !token) throw new Error("Missing Jira credentials.");
    const apiUrl = `${url.replace(/\/$/, '')}/rest/api/2/issue/${jiraId}`;
    const auth = Buffer.from(`${email}:${token}`).toString('base64');
    
    try {
        const res = await axios.get(apiUrl, {
            headers: {
                "Authorization": `Basic ${auth}`,
                "Accept": "application/json"
            }
        });
        
        const summary = res.data.fields.summary || '';
        const desc = res.data.fields.description || '';
        return { summary, description: desc };
    } catch (e: any) {
        throw new Error(`Jira fetch failed: ${e.response?.data?.errorMessages || e.message}`);
    }
}

export async function testJiraConnection(url: string, email: string, token: string) {
    if (!url || !email || !token) return { success: false, msg: "Missing credentials" };
    const apiUrl = `${url.replace(/\/$/, '')}/rest/api/2/myself`;
    const auth = Buffer.from(`${email}:${token}`).toString('base64');
    try {
        await axios.get(apiUrl, { headers: { "Authorization": `Basic ${auth}` } });
        return { success: true, msg: "Connected to Jira!" };
    } catch (e: any) {
        return { success: false, msg: e.message };
    }
}
